content/terraform/v1.12.x/docs/language/manage-sensitive-data.mdx

```diff
@@ -0,0 +1,287 @@
+---
+
+# Manage sensitive data in your configuration
+
+Terraform may require access to sensitive data, such as cloud provider credentials, API tokens, and other secrets to provision your infrastructure. If you add secret values directly to your configuration, Terraform stores those secrets in its state and plan files. Learn about the different options for securing sensitive data in your configuration, and the tradeoffs of each approach.
+
+## Background
+
+Terraform state and plan files contain detailed information about your infrastructure, including resource attributes and metadata that can contain sensitive values, such as initial database passwords or API tokens.
+
+If you are developing with Terraform locally, Terraform stores your state in a plaintext file, which includes any secret values you defined in your configuration. Treat your state file as sensitive data by excluding it from Git workflows and following our recommendations to [secure your state file](#state-security-best-practices).
+
+You can also configure Terraform to store your state and plan files in a [remote location](/terraform/language/state/remote). When you configure Terraform to use remote state storage, it only holds state in memory when actively using it. You can encrypt your state at rest, but the encryption method depends on your specific backend. For example, HCP Terraform automatically encrypts state at rest and protects it with TLS in transit.
+
+## Requirements
+
+The following requirements are tied to specific blocks and arguments you can use to handle sensitive values in your Terraform configuration:
+
+- Use Terraform 0.15 or later to add the `sensitive` argument to `variable` and `output` blocks
+- Use Terraform 1.10 or later to add the `ephemeral` argument to variables and child module outputs, or add the `ephemeral` block
+- Use Terraform 1.11 or later to use a write-only argument on a managed resource
+
+Each provider defines any available `ephemeral` blocks and write-only arguments on their managed resources. Reference the [Terraform registry](https://registry.terraform.io) to learn more about what your provider supports.
+
+## Determine how to handle your sensitive data
+
+To decide how to handle sensitive values in your configuration, start by determining whether you want to hide values from the Terraform CLI and HCP Terraform UI, or prevent Terraform from storing those values entirely.
+
+You can add the `sensitive` argument to `variable` or `output` blocks to make Terraform redact those values from CLI output and in the HCP Terraform UI. Terraform also treats any expressions that reference a sensitive variable or output as inherently sensitive. Learn more [about sensitive variables and outputs](#hide-sensitive-variables-and-outputs).
```

Comment:
```suggestion
You can add the `sensitive` argument to `variable` or `output` blocks so that Terraform redacts those values from CLI output and in the HCP Terraform UI. Terraform also treats any expressions that reference a sensitive variable or output as inherently sensitive. Refer to  [Hide sensitive variables and outputs](#hide-sensitive-variables-and-outputs) for instructions on hiding sensitive data in variables and outputs.
```
Without using the heading as the linked text, readers might think that the link goes to a conceptual page.